docs: enhance dialog component documentation with positioning and sizing details (#3946)
vaadin/docs: articles/components/dialog/index.adoc

Usually, non-modal dialogs should be draggable, so that the user can move them to access the user interface beneath.

> === Position

Reviewer: ```suggestion
== Position
```

Otherwise it doesn't appear in the navigation menu:

![image](https://github.com/user-attachments/assets/652df66d-73f9-48ad-bced-09cf6405c46f)

Author: Thanks!

Dialogs can be positioned in two ways:

. [since:com.vaadin:vaadin@V24.6]##By setting fixed coordinates programmatically##

> . By allowing the user to drag them (see <<Draggable>>)

Reviewer: This gave me the impression that these two ways are mutually exclusive, but that's not the case, if I understand correctly. You can set the initial position using the `top` and `left` properties and still enable dragging, right? The same goes for resizing.

Author: Yeah, both cases can work at the same time. I don't have the same impression, but I can be a bit biased. I can try to improve it to remove any misunderstanding.

Reviewer 2: Not sure if I'm getting a mutually exclusive impression, but it does feel a bit strange to present those as two parallel options. How about something like this instead?

```suggestion
By default, dialogs open in the center of the viewport. [since:com.vaadin:vaadin@V24.6]##A different positioning can be set programmatically##. Dialogs can also be made draggable, allowing the end user to move them around.
```

Reviewer 2: (not sure I got the version tag syntax right there, so please check)

Author: It's working:
<img width="751" alt="image" src="https://github.com/user-attachments/assets/15e8353d-2461-4ea9-bd65-cd86d1b5237e">

endif::[]
--

It's recommended to make non-modal dialogs draggable, so that the user can interact with content that might otherwise be obscured by the Dialog. For example, a Dialog for taking notes or for adding widgets to a dashboard by dragging can offer a better experience by allowing the user to move the Dialog around.

Modal dialogs don't benefit from being draggable, as their modality curtain (the dark overlay behind the dialog) obscures the underlying user interface.

== Size

> The Dialog size can be set with the width and height on the Dialog itself (__since V24.6 for React and Lit__). You can also set the size of the content of Dialog, whereby the Dialog scales to accommodate it In both cases, the Dialog can also be made resizable.

Author: Missing dot:
```suggestion
The Dialog size can be set with the width and height on the Dialog itself (__since V24.6 for React and Lit__). You can also set the size of the content of Dialog, whereby the Dialog scales to accommodate it. In both cases, the Dialog can also be made resizable.
```

Modal dialogs don't benefit from being draggable, as their modality curtain (the dark overlay behind the dialog) obscures the underlying user interface.

== Size

> The Dialog size can be controlled in two ways:
> 
> . By setting fixed dimensions using the width and height properties (__since V24.6 for React and Lit__)
> . By making the dialog resizable (see <<Resizable>>)

Reviewer 2: A bit same as above, feels like resizability is presented as an alternative. Also, I think it's best to mention content-based sizing already at the top, like so:

```suggestion
The Dialog size can be set in two ways:

. By setting width and height on the Dialog itself (__since V24.6 for React and Lit__)
. By setting the size of the Dialog's contents, whereby the Dialog scales to accommodate it

In both cases, the Dialog can also be made resizable.
```

Author: In this case, the paragraph about content-based sizing after the code snippets should be removed, right?

Author: NVM. I just saw the comment below.

[.example]
--

ifdef::lit[]
[source,html]
----
<source-info group="Lit"></source-info>
<vaadin-dialog width="400px" height="200px">...</vaadin-dialog>
----
endif::[]

ifdef::flow[]
[source,java]
----
<source-info group="Flow"></source-info>
Dialog dialog = new Dialog();
dialog.setWidth("400px");
dialog.setHeight("200px");
----
endif::[]

ifdef::react[]
[source,html]
----
<source-info group="React"></source-info>
<Dialog width="400px" height="200px">...</Dialog>
----
endif::[]
--

> The dialog's default size adjusts automatically based on its content. Fixed dimensions could be used when you need precise control over the dialog's size, for example, when displaying a form or grid with specific space requirements.

Reviewer 2: Then we can remove this paragraph.

> 
> The dialog's default size adjusts automatically based on its content. Fixed dimensions could be used when you need precise control over the dialog's size, for example, when displaying a form or grid with specific space requirements.

Author: ```suggestion
```

=== Resizable

> A resizable dialog allows the user to resize the Dialog by dragging from the edges of the Dialog with a pointing device.
> Dialogs aren't resizable by default.

Reviewer 2: Let's rewrite this weird old sentence while we're at it 😅 

```suggestion
The Dialog can be configured to allow the end user to resize it by dragging from its edges.
```
